contracts/src/v0.8/vrf/VRFV2PlusWrapper.sol

```diff
@@ -0,0 +1,521 @@
+
+  // Configuration fetched from VRFCoordinatorV2
+
+  // s_configured tracks whether this contract has been configured. If not configured, randomness
+  // requests cannot be made.
+  bool public s_configured;
+
+  // s_disabled disables the contract when true. When disabled, new VRF requests cannot be made
+  // but existing ones can still be fulfilled.
+  bool public s_disabled;
+
+  // s_fallbackWeiPerUnitLink is the backup LINK exchange rate used when the LINK/NATIVE feed is
+  // stale.
+  int256 private s_fallbackWeiPerUnitLink;
+
+  // s_stalenessSeconds is the number of seconds before we consider the feed price to be stale and
+  // fallback to fallbackWeiPerUnitLink.
+  uint32 private s_stalenessSeconds;
+
+  // s_fulfillmentFlatFeeLinkPPM is the flat fee in millionths of LINK that VRFCoordinatorV2
+  // charges.
+  uint32 private s_fulfillmentFlatFeeLinkPPM;
+
+  // s_fulfillmentFlatFeeLinkPPM is the flat fee in millionths of LINK that VRFCoordinatorV2
+  // charges.
+  uint32 private s_fulfillmentFlatFeeEthPPM;
+
+  // Other configuration
+
+  // s_wrapperGasOverhead reflects the gas overhead of the wrapper's fulfillRandomWords
+  // function. The cost for this gas is passed to the user.
+  uint32 private s_wrapperGasOverhead;
+
+  // s_coordinatorGasOverhead reflects the gas overhead of the coordinator's fulfillRandomWords
+  // function. The cost for this gas is billed to the subscription, and must therefor be included
+  // in the pricing for wrapped requests. This includes the gas costs of proof verification and
+  // payment calculation in the coordinator.
+  uint32 private s_coordinatorGasOverhead;
+
+  // s_wrapperPremiumPercentage is the premium ratio in percentage. For example, a value of 0
+  // indicates no premium. A value of 15 indicates a 15 percent premium.
+  uint8 private s_wrapperPremiumPercentage;
+
+  // s_keyHash is the key hash to use when requesting randomness. Fees are paid based on current gas
+  // fees, so this should be set to the highest gas lane on the network.
+  bytes32 s_keyHash;
+
+  // s_maxNumWords is the max number of words that can be requested in a single wrapped VRF request.
+  uint8 s_maxNumWords;
+
+  struct Callback {
+    address callbackAddress;
+    uint32 callbackGasLimit;
+    uint256 requestGasPrice;
+  }
+  mapping(uint256 => Callback) /* requestID */ /* callback */ public s_callbacks;
+
+  constructor(
+    address _link,
+    address _linkEthFeed,
+    address _coordinator
+  ) ConfirmedOwner(msg.sender) VRFConsumerBaseV2Plus(_coordinator, msg.sender) {
+    if (_link != address(0)) {
+      s_link = LinkTokenInterface(_link);
+    }
+    if (_linkEthFeed != address(0)) {
+      s_linkEthFeed = AggregatorV3Interface(_linkEthFeed);
+    }
+    COORDINATOR = ExtendedVRFCoordinatorV2PlusInterface(_coordinator);
+
+    // Create this wrapper's subscription and add itself as a consumer.
+    uint64 subId = ExtendedVRFCoordinatorV2PlusInterface(_coordinator).createSubscription();
```

Comment:
this logic may need changes once we finalize upgradeability e.g. as it currently stands, subscription manager will need to be owner of this contract. `subId` and `coordinator` will need to be mutable but we can follow up on this change in future PRs

Reply:
Fair enough, good callout for now.

Reply:
I think its fine to redeploy the wrapper. Unless we find that this approach is not agreeable, lets not complicate the wrapper more than it needs to be complicated due to the native token changes.